### PR TITLE
[No QA] Update PR template to be more explicit about expectations to check off everything

### PR DESCRIPTION
**Pull request by flodnv**
### Details

Redo of https://github.com/Expensify/App/pull/9597
I have seen some confusions, hopefully this clears them up.
cc @Expensify/contributor-plus 

### Fixed Issues
https://expensify.slack.com/archives/C01SKUP7QR0/p1655892677571859

### Tests

none needed

### PR Review Checklist

this is what I am modifying 

### QA Steps

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -99,17 +100,18 @@ This is a checklist for PR authors & reviewers. Please make sure to complete all
...
+    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
     - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
 - [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
 - [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
+- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
```
Comment (flodnv) [inline .github/PULL_REQUEST_TEMPLATE.md:114]: This resolves https://github.com/Expensify/App/pull/9597#discussion_r908724369

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -74,17 +74,18 @@ This is a checklist for PR authors & reviewers. Please make sure to complete all
...
+    - [ ] A similar style doesn't already exist
+    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
 - [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
 - [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
-
+- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.
 
 <details>
-<summary><h4>PR Reviewer Checklist</h4></summary>
+<summary><h4>PR Reviewer Checklist to copy/paste into a new comment after the author checklist is completed, and to be completed by a Contributor+</h4></summary>
```
Comment (chiragsalian) [inline .github/PULL_REQUEST_TEMPLATE.md:86]: you've update this to mention here that its to be completed by C+. I thought even we have to complete this when reviewing. Can you clarify? If its just C+ lets rename the section to be `C+ Reviewer Checklist` so there is no ambiguity. If its both let's rephrase to mention to be complete by all reviewers.

Comment (chiragsalian) [reply .github/PULL_REQUEST_TEMPLATE.md:86]: Also this summary is so long that the dropdown comes out wierdly. 
image

Can we split it up such that copy/paste onwards is on the next line

**Review by iwiznia — DISMISSED** [review]

**Review by sketchydroide — DISMISSED** [review]

**Review by puneetlath — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (flodnv) [reply .github/PULL_REQUEST_TEMPLATE.md:86]: 1. The discussion concluded with what I understood: it's just the C+. However, it also highlighted that a PR Reviewer may not be a C+. As such, I think the current wording is correct.
2. Bump?

**Review by puneetlath — APPROVED** [review]

**Review by chiragsalian — APPROVED** [review]

**Comment by chiragsalian** [comment]
Going to merge since most of the people approved before and the most recent update was a minor change.

**Pull request merged**